Sources/HTTPTypes/HTTPFields.swift

```diff
@@ -34,77 +24,33 @@ public struct HTTPFields: Sendable, Hashable {
     private final class _Storage: @unchecked Sendable, Hashable {
         var fields: [(field: HTTPField, next: UInt16)] = []
         var index: [String: (first: UInt16, last: UInt16)]? = [:]
-        #if canImport(os.lock)
-        let lock = UnsafeMutablePointer<os_unfair_lock>.allocate(capacity: 1)
-        #else
-        let lock = UnsafeMutablePointer<pthread_mutex_t>.allocate(capacity: 1)
-        #endif
-
-        init() {
-            #if canImport(os.lock)
-            self.lock.initialize(to: os_unfair_lock())
-            #else
-            let err = pthread_mutex_init(self.lock, nil)
-            precondition(err == 0, "pthread_mutex_init failed with error \(err)")
-            #endif
-        }
-
-        deinit {
-            #if !canImport(os.lock)
-            let err = pthread_mutex_destroy(self.lock)
-            precondition(err == 0, "pthread_mutex_destroy failed with error \(err)")
-            #endif
-            self.lock.deallocate()
-        }
+        let lock = LockStorage.create(value: ())
```

Comment:
`LockStorage` is a class so it already gives us reference semantics. We should be able to drop the `_Storage` class here now and make it a struct. Then put that struct into the `LockStorage` that should save us an allocation.

Reply:
If we do that, both `fields` and `index` need to be a part of LockStorage. `lock` currently only protects `index` in the situation where there are multiple `HTTPFields` sharing the same storage. Can I add an unsafe accessor from LockStorage for `fields` access?

Reply:
I see. I didn't know we just protected one. I think it's fine if we leave it as is in the PR. We can revisit later if we find that allocation problematic.